src/nativeAssetManager.js

```diff
@@ -333,17 +336,24 @@ export function newNativeAssetManager(win, pubUrl) {
   /**
    * Replaces occurrences of native placeholder values with their actual values
    * in the given document.
+   * If there are still native placeholders in the template that were not replaced (because no asset value was sent),
+   * those placeholders will be replaced by an empty string.
    */
   function replace(document, { assets, adId }) {
     let html = document;
 
     (assets || []).forEach(asset => {
-      const flag = (typeof win.pbNativeData !== 'undefined');
+      const flag = pbNativeDataHasValidType();
       const searchString = (adId && !flag) ? `${NATIVE_KEYS[asset.key]}:${adId}` : ((flag) ? '##'+`${NATIVE_KEYS[asset.key]}`+'##' : `${NATIVE_KEYS[asset.key]}`);
       const searchStringRegex = new RegExp(searchString, 'g');
       html = html.replace(searchStringRegex, asset.value);
     });
 
+    scanForPlaceholders().forEach(placeholder => {
+      const searchString = pbNativeDataHasValidType() ? `##${placeholder}##` : placeholder;
```

Comment:
I'm not sure this is correct, because the logic on how to construct the placeholder is different from what's just a few lines above in `replace`. (Unfortunately I'm not sure what the logic _should_ be either, it looks like it supports yet more obscure use cases).

Would it be better to move this inside `replace`? on line 345 you could loop over `scanForPlaceHolders()` instead of `assets`.

Reply:
@dgirardi I get your point!

I flipped it like you suggested, now looping over `scanForPlaceholders` with more or less the original logic. Is this what you were thinking of?